arangod/Pregel/Worker/WorkerConfig.cpp

```diff
@@ -104,12 +102,22 @@ VertexID WorkerConfig::documentIdToPregel(std::string_view documentID) const {
   std::string_view collPart = docRef.substr(0, pos);
   std::string_view keyPart = docRef.substr(pos + 1);
 
-  ShardID responsibleShard;
-
-  auto& ci = _vocbase->server().getFeature<ClusterFeature>().clusterInfo();
-  ResolveShard::resolve(ci, this, std::string(collPart),
-                        StaticStrings::KeyString, keyPart, responsibleShard);
-
-  PregelShard source = this->shardId(responsibleShard);
-  return VertexID(source, std::string(keyPart));
+  if (!ServerState::instance()->isRunningInCluster()) {
+    return VertexID(shardId(std::string{collPart}), std::string(keyPart));
+  } else {
+    VPackBuilder partial;
+    partial.openObject();
+    partial.add(
+        StaticStrings::KeyString,
```

Comment:
I know this has been implemented like this before, but we should add a follow-up ticket to inspect why the `StaticStrings::KeyString` is being used here and not the "shardKeyAttribute".